FileManager.WPF/FileManager.WPF/Model/BaseFile.cs

```diff
@@ -0,0 +1,78 @@
+﻿
+using NLog;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+
+namespace FileManager.WPF.Model
+{
+    /// <summary> Класс описывающий базовую модель файлов и дерикторий. </summary>
+    public class BaseFile : INotifyPropertyChanged
+    {
+        protected internal static ILogger _logger;
+        protected internal string _fullPath;
+        protected internal string _name;
+        private string[] _fileInfo;
+        private string _imagePath;
+
+        #region feilds
+
+        /// <summary> Путь к иконке файла/папки. </summary>
+        public string ImagePath { get => _imagePath; set => _imagePath = value; }
```

Comment:
Если это модель-представления, то в сеттере этого свойства (и свойств ниже) не генерируется событие изменения этого свойства.
Надо было написать 
```suggestion
        public string ImagePath { get => _imagePath; set => Set(ref _imagePath , value); }
```
Реализация метода `Set` ниже